pandas/tests/series/test_asof.py

```diff
@@ -148,3 +148,9 @@ def test_errors(self):
         s = Series(np.random.randn(N), index=rng)
         with self.assertRaises(ValueError):
             s.asof(s.index[0], subset='foo')
+
+    def test_all_nans(self):
+        # series is all nans
+        result = Series([np.nan]).asof([0])
```

Comment:
Can you also add a case not using zero as the argument? 
And can you also add the case of a scalar, and of multiple values? (eg `s.asof(10)` and `s.asof([10, 11])`